### PR TITLE
created and tested PATCH /api/article/:article_id endpoint to increme…

### PR DESCRIPTION
**Pull request by theshortsmonkey**
…nt votes with asscoiated error handling

### CODE REVIEW

File: controllers/articles.controllers.js [hunk 1]
```diff
@@ -34,4 +34,15 @@ exports.postCommentByArticleId = (req,res,next) => {
     res.status(201).send({postedComment})
   })
   .catch(next)
+}
+
+exports.patchArticleById = (req,res,next) => {
+  const {article_id} = req.params
+  const promises = [updateArticleById(article_id,req.body),selectArticleyById(article_id)]
```
Comment (nc-hannah-burks) [inline controllers/articles.controllers.js:41]: bit unsure why we are doing selectArticleyById here? We are only interested in the returned updated row from updateArticleById

File: models/articles.models.js [hunk 2]
```diff
@@ -68,3 +68,16 @@ exports.insertCommentByArticleId = (article_id, comment) => {
       return rows[0]
     })
 }
+
+exports.updateArticleById = (article_id,body) => {
+  const {inc_votes} = body
+  if (!inc_votes || (typeof inc_votes !== 'number')) {
```
Comment (nc-hannah-burks) [inline models/articles.models.js:74]: We don't need to check if inc_votes is a number, as if it is not, PSQL will throw an error for this.

File: models/articles.models.js [hunk 3]
```diff
@@ -68,3 +68,16 @@ exports.insertCommentByArticleId = (article_id, comment) => {
...
+    return Promise.reject({status:400,customErrMsg:'invalid vote increment supplied'})
+  }
+  return db.query(`UPDATE articles
+  SET votes = votes + $1
+  WHERE article_id = $2 RETURNING *;`,[inc_votes,article_id])
+  .then(({rows}) => {
```
Comment (nc-hannah-burks) [inline models/articles.models.js:80]: We want a custom error here to handle a 404 (non existent ID, e.g. 0 or 9999 - if we get no updated row back, what does that tell us?)

File: __tests__/app.test.js [hunk 4]
```diff
@@ -86,6 +86,82 @@ describe('/api/articles/:article_id endpoint', () => {
...
+      author: 'butter_bridge',
+      body: 'I find this existence challenging',
+      created_at: '2020-07-09T20:11:00.000Z',
+      votes: 105,
+      article_img_url:
+        'https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700',
+    }
+    return request(app)
+    .patch('/api/articles/1')
+    .send({inc_votes:5,article_id:'test'})
```
Comment (nc-hannah-burks) [inline __tests__/app.test.js:123]: Excellent - well done !

File: controllers/articles.controllers.js (hunk 1 above)
Comment (nc-hannah-burks) [reply controllers/articles.controllers.js:41]: I think you are using it to handle the 404, however, we could do that by simply checking the updated rows

File: __tests__/app.test.js [hunk 5]
```diff
@@ -86,6 +86,82 @@ describe('/api/articles/:article_id endpoint', () => {
...
+    }
+    return request(app)
+    .patch('/api/articles/1')
+    .send({inc_votes:5})
+    .expect(200)
+    .then(({body}) => {
+      expect(body.updatedArticle).toEqual(expectedArticle)
+    })
+  })
+  test("PATCH: 200 should increment the votes of a specified article by the supplied amount,ignoring unnessecary properties in requested body", () => {
```
Comment (nc-hannah-burks) [inline __tests__/app.test.js:109]: It might be worth also adding a test if trying to update votes with a negative number. This should also work {inc_votes: -100}

File: controllers/articles.controllers.js (hunk 1 above)
Comment (theshortsmonkey) [reply controllers/articles.controllers.js:41]: I found  I was using the same code multiple times in the models so I refactored to this way to keep the code drier.

File: models/articles.models.js (hunk 3 above)
Comment (theshortsmonkey) [reply models/articles.models.js:80]: I am handling this with the selectArticleById in the Promise.all in the controller

File: models/articles.models.js (hunk 2 above)
Comment (theshortsmonkey) [reply models/articles.models.js:74]: this throws the same PSQL error that I captured previously with a specific 'invalid id supplied' error message. As the PSQL error is generic for 'not integer'  I wanted to differentiate error messages so included this check here.
